iroh-gateway/src/response.rs

```diff
@@ -12,97 +10,158 @@ pub const ERR_UNSUPPORTED_FORMAT: &str = "unsuported format";
+            "application/vnd.ipld.car" | "car" => Ok(ResponseFormat::Car),
+            "fs" | "" => Ok(ResponseFormat::Fs(String::new())),
+            rf => {
+                if rf.starts_with("application/vnd.ipld.") {
+                    Ok(ResponseFormat::Fs(rf.to_string()))
+                } else {
+                    Err(format!("{}: {}", ERR_UNSUPPORTED_FORMAT, rf))
+                }
+            }
         }
     }
 }
 
 impl ResponseFormat {
-    pub fn write_headers(&self, headers: &mut HashMap<String, String>) {
+    pub fn write_headers(&self, headers: &mut HeaderMap) {
         match self {
             ResponseFormat::Raw => {
-                headers.insert(CONTENT_TYPE.to_string(), CONTENT_TYPE_IPLD_RAW.to_string());
-                headers.insert(
-                    HEADER_X_CONTENT_TYPE_OPTIONS.to_string(),
-                    VALUE_XCTO_NOSNIFF.to_string(),
-                );
+                headers.insert(CONTENT_TYPE, CONTENT_TYPE_IPLD_RAW.clone());
+                headers.insert(&HEADER_X_CONTENT_TYPE_OPTIONS, VALUE_XCTO_NOSNIFF.clone());
             }
             ResponseFormat::Car => {
-                headers.insert(CONTENT_TYPE.to_string(), CONTENT_TYPE_IPLD_CAR.to_string());
-                headers.insert(
-                    HEADER_X_CONTENT_TYPE_OPTIONS.to_string(),
-                    VALUE_XCTO_NOSNIFF.to_string(),
-                );
+                headers.insert(CONTENT_TYPE, CONTENT_TYPE_IPLD_CAR.clone());
+                headers.insert(&HEADER_X_CONTENT_TYPE_OPTIONS, VALUE_XCTO_NOSNIFF.clone());
+                headers.insert(ACCEPT_RANGES, VALUE_NONE.clone());
+                headers.insert(CACHE_CONTROL, VALUE_NO_CACHE_NO_TRANSFORM.clone());
             }
-            ResponseFormat::Html => {
-                headers.insert(CONTENT_TYPE.to_string(), CONTENT_TYPE_HTML.to_string());
+            ResponseFormat::Fs(_) => {
+                headers.insert(CONTENT_TYPE, CONTENT_TYPE_OCTET_STREAM.clone());
+            }
+        }
+    }
+
+    pub fn get_extenstion(&self) -> String {
+        match self {
+            ResponseFormat::Raw => "bin".to_string(),
+            ResponseFormat::Car => "car".to_string(),
+            ResponseFormat::Fs(s) => {
+                if s.is_empty() {
+                    String::new()
+                } else {
+                    s.split('.').last().unwrap().to_string()
+                }
             }
-            ResponseFormat::Fs => {
-                headers.insert(
-                    CONTENT_TYPE.to_string(),
-                    CONTENT_TYPE_OCTET_STREAM.to_string(),
-                );
+        }
+    }
+
+    pub fn try_from_headers(headers: &HeaderMap) -> Result<Self, String> {
+        if headers.contains_key("Accept") {
+            let h_values = headers.get("Accept").unwrap().to_str().unwrap();
+            let h_values = h_values.split(',').collect::<Vec<&str>>();
```

Comment:
no need to collect if you iterate immediately again

Reply:
You can avoid one unwrap() with something like:
```rust
  if let Some(h_values) = headers.get("Accept") {
  ...
}
```